tcp/tcp.go

```diff
@@ -261,7 +261,7 @@ func (p *Proxy) handleConn(conn net.Conn) error {
 			if IsSocketClosed(err) && connectionClosed.Load().(bool) {
 				return
 			}
-			if err == io.EOF {
+			if errors.Is(err, io.EOF) {
```

Comment:
It's not necessarily reported, but can you consider the relatively small update in #4203 ; would love to gate some unnecessary logging calls in this package for expected errors; the change may be cherry picked onto this PR, with only some small test update necessary for a recent go version. (maintenance). two birds one stone?